src/android/FirebaseMessagingRouterService.java

```diff
@@ -21,4 +21,16 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
             intercomPushClient.handlePush(getApplication(), remoteMessage.getData());
         }
     }
+
+    @Override
+   public void onNewToken(String token) {
+       // Send a new token to Pushwoosh
+       PushwooshFcmHelper.onTokenRefresh(this, token);
+       sendTokenToAnotherService(token);
+   }
+
+   private void sendTokenToAnotherService(String token) {
```

Comment:
I don't really feel that this deserves its own function anyway - you can just send it to Intercom in `onNewToken()` like we're doing above in `onMessageReceived()`. (After all, you're not putting `PushwooshFcmHelper.onTokenRefresh()` in its own method.)